### PR TITLE
Support multi directories and fallbacks with wildcards configuration

### PR DESCRIPTION
**Pull request by kclang**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
 app.use('/de', express.static(__dirname + '/www'));
 ```
 
+## Multiple Directories support
+
+i18n-node support a multiple directory usage. This option let you register different directories to provide translations. Just add "multiDirectories : true" in your configuration then use the configure method to provide new directory and a name for it. Finally call translations methods with directory name as last argument.
+Please note that in case you do not use the updateFiles you may ommit directory name and juste call translations method as usual.
```
Comment (sngo25) [inline README.md:114]: juste => just
translations method => translation methods

File: README.md [hunk 2]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
 app.use('/de', express.static(__dirname + '/www'));
 ```
 
+## Multiple Directories support
+
+i18n-node support a multiple directory usage. This option let you register different directories to provide translations. Just add "multiDirectories : true" in your configuration then use the configure method to provide new directory and a name for it. Finally call translations methods with directory name as last argument.
```
Comment (sngo25) [inline README.md:113]: `multiDirectories : true` should be wrapped in code block

File: README.md [hunk 3]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
 app.use('/de', express.static(__dirname + '/www'));
 ```
 
+## Multiple Directories support
```
Comment (sngo25) [inline README.md:111]: Multiple directories support

File: README.md (hunk 2 above)
Comment (sngo25) [inline README.md:113]: multiple directories usage

File: README.md [hunk 4]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+
+i18n-node support a multiple directory usage. This option let you register different directories to provide translations. Just add "multiDirectories : true" in your configuration then use the configure method to provide new directory and a name for it. Finally call translations methods with directory name as last argument.
+Please note that in case you do not use the updateFiles you may ommit directory name and juste call translations method as usual.
+
+### Exemple
```
Comment (sngo25) [inline README.md:116]: Example

File: README.md [hunk 5]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+
+Configure i18n as usual. Add "multiDirectories" as true, specify new directory path and a name (if need).
```
Comment (sngo25) [inline README.md:118]: `multiDirectories` should be inside code block

File: README.md [hunk 6]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+
+	i18n.configure({
+      locales:['en', 'de'],
```
Comment (sngo25) [inline README.md:121]: 1/ bad indentation
2/ please wrap this in JS code block

File: README.md [hunk 7]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+```
+__('sentence to translate');
+//Same as __('sentence to translate', 'default');
+
+__('sentence to translate', 'newDirectory');
+```
+
+### Details
+
+When you set multiDirectories at true in the configure method, a "directories" object is initialized. Then every time you register a new source directory with configure method you extend this object with a new directory path and a name for this path. Finally you can call translations methods and provide as last argument the name of the directory path you set earlier.
```
Comment (sngo25) [inline README.md:138]: `multiDirectories`

File: README.md [hunk 8]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+
+Note that when you activate the multiDirectories option you cannot reset every option for every directory. It means you must have the same directory configuration (eg : indentation, object notation, prefixes ...).
```
Comment (sngo25) [inline README.md:140]: `multiDirectories`

Comment (sngo25) [inline README.md:140]: no space before `:` and `...`

File: README.md [hunk 9]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+
+The available option are :
```
Comment (sngo25) [inline README.md:142]: no space before `:`

File: README.md [hunk 10]
```diff
@@ -108,6 +108,46 @@ app.use('/en', express.static(__dirname + '/www'));
...
+
+ - directory : path to sources
+ - dirName : directory name / module name
+ - locales
+ - fallbacks
+
+__If you want to reset your i18n configuration__, call configure() with multiDirectories at false and every option will be reseted
```
Comment (sngo25) [inline README.md:149]: should not be bolded

File: README.md [hunk 11]
```diff
@@ -138,7 +178,8 @@ i18n.configure({
     locales:['en', 'de'],
 
     // fall back from Dutch to German
```
Comment (sngo25) [inline README.md:180]: is this comment still useful?

File: package.json [hunk 12]
```diff
@@ -23,7 +23,8 @@
     "math-interval-parser": "^1.1.0",
     "messageformat": "^0.3.1",
     "mustache": "*",
-    "sprintf-js": ">=1.0.3"
+    "sprintf-js": ">=1.0.3",
+    "extend": "*"
```
Comment (sngo25) [inline package.json:27]: the version should be fixed

File: test/i18n.configure.js [hunk 13]
```diff
@@ -2,44 +2,123 @@ var i18n = require('../i18n'),
...
     should.exist(stats);
     if (stats) {
       try {
         fs.unlinkSync('./customlocales/customprefix-de.customextension');
         fs.unlinkSync('./customlocales/customprefix-en.customextension');
         fs.rmdirSync('./customlocales');
-      } catch (e) {}
+      } catch (e) {
+
+      }
```
Comment (sngo25) [inline test/i18n.configure.js:108]: please revert

File: test/i18n.configure.js [hunk 14]
```diff
@@ -2,44 +2,123 @@ var i18n = require('../i18n'),
...
+  it('should be possible to setup a custom directory', function () {
     var stats = fs.lstatSync('./customlocales');
     should.exist(stats);
   });
 
-  it('should be possible to read custom files with custom prefixes and extensions', function() {
+  it('should be possible to read custom files with custom prefixes and extensions', function () {
     var statsde = fs.lstatSync('./customlocales/customprefix-de.customextension'),
-      statsen = fs.lstatSync('./customlocales/customprefix-en.customextension');
+    statsen = fs.lstatSync('./customlocales/customprefix-en.customextension');
```
Comment (sngo25) [inline test/i18n.configure.js:120]: bad indentation

File: test/i18n.configureAutoreload.js [hunk 15]
```diff
@@ -12,14 +12,75 @@ function reconfigure(config) {
...
   i18n.configure(config);
 }
+describe('autoreload configuration with multiDirectories enabled', function() {
+
+  var testScope = {};
+  var directory = path.resolve(__dirname + '/../testlocalesauto');
+  var directory2 = path.resolve(__dirname + '/../testlocalesauto2');
+
+  it('will start with empty catalogs', function(done) {
+    try{
```
Comment (sngo25) [inline test/i18n.configureAutoreload.js:22]: missing space before {

File: test/i18n.configureAutoreload.js [hunk 16]
```diff
@@ -12,14 +12,75 @@ function reconfigure(config) {
...
 describe('autoreload configuration', function() {
 
   var testScope = {};
   var directory = path.resolve(__dirname + '/../testlocalesauto');
 
   it('will start with empty catalogs', function(done) {
-    fs.mkdirSync(directory);
+    try {
+        fs.mkdirSync(directory);
+    }catch(e) {}
```
Comment (sngo25) [inline test/i18n.configureAutoreload.js:83]: missing space after }

File: README.md (hunk 11 above)
Comment (kclang) [reply README.md:180]: yes, it is

**Comment by kclang** [comment]
replace this PR by 2 others: https://github.com/linagora/i18n-node/pull/5 and https://github.com/linagora/i18n-node/pull/6

**Pull request closed without merging**